classes/shaman.lua

```diff
@@ -133,6 +183,58 @@ local function registerClass(self)
         self:RegisterAura("elemental_focus_1", 1, 16246, "echo_of_the_elements", "Left", 1, 255, 255, 255, true);
         self:RegisterAura("elemental_focus_2", 2, 16246, "echo_of_the_elements", "Left + Right (Flipped)", 1, 255, 255, 255, true);
     end
+    
+    if self.IsCata() then
+        self:RegisterAura("elemental_focus_1", 1, 16246, "genericarc_05", "Left", 1.5, 255, 255, 255, true);
+        self:RegisterAura("elemental_focus_2", 2, 16246, "genericarc_05", "Left + Right (Flipped)", 1.5, 255, 255, 255, true);
+        local lavaBurstCata = 51505;
+        self:RegisterAura("lava_surge", 0, lavaBurstCata, "imp_empowerment", "Left + Right (Flipped)", 1, 255, 255, 255, true, {lavaBurstCata}, true);
+        self:RegisterCounter("lava_surge");
+        if (not RollingThunderHandler.initialized) then
+            RollingThunderHandler:init();
+        end
+        for lightningShieldStacks=6,9 do
+            local auraName = "fulmination_"..lightningShieldStacks;
+            local scale = 0.5 + 0.1 * (lightningShieldStacks - 5); -- 60%, 70%, 80%
+            local pulse = lightningShieldStacks == 9;
+            self:RegisterAura(auraName, lightningShieldStacks, RollingThunderHandler.fakeSpellID, "fulmination", "Top", scale, 255, 255, 255, pulse, RollingThunderHandler.earthShockSpells);
+        end
+        -- Maelstrom Weapon
+        local lightningBolt = 403;
+        local chainLightning = 421;
+        local healingSurge = 8004;
+        local healingWave = 331;
+        local chainHeal = 1064;
+        local greaterHealingWave = 77472;
+        local healingRain = 73920;
+        local hex = 51514;
+        local maelstromSpells = {
+            GetSpellInfo(lightningBolt),
+            GetSpellInfo(chainLightning),
+            GetSpellInfo(healingSurge),
+            GetSpellInfo(greaterHealingWave),
+            GetSpellInfo(healingWave),
+            GetSpellInfo(chainHeal),
+            GetSpellInfo(healingRain),
+            GetSpellInfo(hex),
+        }
+        self:RegisterAura("maelstrom_weapon_1", 1, 53817, "maelstrom_weapon_1", "Top", 1, 255, 255, 255, false);
+        self:RegisterAura("maelstrom_weapon_2", 2, 53817, "maelstrom_weapon_2", "Top", 1, 255, 255, 255, false);
+        self:RegisterAura("maelstrom_weapon_3", 3, 53817, "maelstrom_weapon_3", "Top", 1, 255, 255, 255, false);
+        self:RegisterAura("maelstrom_weapon_4", 4, 53817, "maelstrom_weapon_4", "Top", 1, 255, 255, 255, false);
+        self:RegisterAura("maelstrom_weapon_5", 5, 53817, "maelstrom_weapon", "Top", 1, 255, 255, 255, true, maelstromSpells);
+
+        -- Tidal Waves
+        local tidalSpells = {
+            GetSpellInfo(greaterHealingWave),
+            GetSpellInfo(healingWave),
+            GetSpellInfo(healingSurge),
+        }

```

Comment:
Because `GetSpellInfo` returns multiple values, your table will be polluted by the last call.

If you want to have fun, you can type this command to your chat in-game, you'll see what happens:
```lua
/dump { GetSpellInfo(77472), GetSpellInfo(331), GetSpellInfo(8004) }
```

I would suggest enclosing `GetSpellInfo` calls in parentheses, for example:
```lua
local tidalSpells = {
    (GetSpellInfo(greaterHealingWave)),
    (GetSpellInfo(healingWave)),
    (GetSpellInfo(healingSurge)),
}
```
This will force Lua to capture the first return value and nothing else.